File: Arrays/4.MedianofTwoSortedArrays.c
double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size) {
    int n = nums1Size + nums2Size;
    int res[n];
    int i = 0, j = 0, k = 0;
    while(i < nums1Size && j < nums2Size){
        if(nums1[i] <= nums2[j]){
            res[k] = nums1[i];
            i++;
            k++;
        }
        else{
            res[k] = nums2[j];
            j++;
            k++;
        }
    }
    while(i < nums1Size){
        res[k++] = nums1[i++];
    }
    while(j < nums2Size){
        res[k++] = nums2[j++];
    }
    double median = 0;
    if(n % 2 == 0){
        int m = n/2;                     
        median = (res[m-1] + res[m])/2.0;
        return median;            
    }
    else{
        int m = n/2;
        median = res[m];
        return median;
    }
}

File: Arrays/4. Median of Two Sorted Arrays.java
class Solution {
    public double findMedianSortedArrays(int[] nums1, int[] nums2) {
        int [] newarr = new int [nums1.length + nums2.length];
        int i = 0, j = 0, k = 0;
        while (i < nums1.length && j < nums2.length){ 
            if(nums1[i] > nums2[j]){
                newarr[k++] = nums2[j++];
            }
            else{
                newarr[k++] = nums1[i++];
            }
        }
        while(i < nums1.length){
            newarr[k++] = nums1[i++];
        }
        while(j < nums2.length){
            newarr[k++] = nums2[j++];
        }
        double median = 0;
        int n = newarr.length/2;
        if(newarr.length % 2 == 0){
            median = (newarr[n-1] + newarr[n])/2.0;
        }
        else{
            median = (newarr[n]);
        }
        return median;
    }
}
